### PR TITLE
[ZEPPELIN-2612] Remove duplicates from ZeppelinConfiguration

### PR DESCRIPTION
**Pull request by DrIgor**
### What is this PR for?
Remove duplicated code from `ZeppelinConfiguration` and make proper use of commons-configuraion

And I propose using interpolation instead of `getRelativeDir` (Not in this PR).
For example,
```
ZEPPELIN_INTERPRETER_DIR("zeppelin.interpreter.dir", "${zeppelin.home}/interpreter")
```

### What type of PR is it?
[Refactoring]

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-2612

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
@DrIgor This is not a trivial change, please create a ticket and describe the purpose.

File: zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java [hunk 1]
```diff
@@ -171,7 +172,6 @@ public ZeppelinServer() throws Exception {
   public static void main(String[] args) throws InterruptedException {
 
     ZeppelinConfiguration conf = ZeppelinConfiguration.create();
-    conf.setProperty("args", args);
```
Comment (DrIgor) [inline zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java]: Usage was removed in https://github.com/apache/zeppelin/commit/0330452231c9dae85bb642fbd7a8b9cc16c6777d#diff-4bd397174ee0f21a8d3b788b97bc8711L487

File: pom.xml [hunk 2]
```diff
@@ -109,6 +109,7 @@
 
     <!-- test library versions -->
     <junit.version>4.12</junit.version>
+    <hamcrest-all.version>1.3</hamcrest-all.version>
```
Comment (felixcheung) [inline pom.xml]: why is this needed?

Comment (felixcheung) [inline pom.xml]: if this is testing only, should it go to submodule pom file?
also we need to add this to LICENSE

Comment (DrIgor) [inline pom.xml]: `jUnit` depends on `hamcrest-core 1.3` and we already used it in the project. I placed it right after jUnit.
